src/groupeddataframe/complextransforms.jl

```diff
@@ -64,23 +67,26 @@ function _combine_with_first(first::Union{NamedTuple, DataFrameRow, AbstractData
     return idx, outcols, collect(Symbol, finalcolnames)
 end
 
+function _names_match(r1, t::NTuple{N, Symbol}) where N
+    n1 = _getnames(r1)
```

Comment:
Maybe have the function take names directly? Than would avoid specializing it on different row types.

Reply:
OK

Reply:
But wait - different row types will produce different `_getnames(r1)` type anyway, so I do not think it is much benefit of this.

Reply:
Don't e.g. `DataFrameRow` and `AbstractDataFrame` give the same type?

Reply:
No, `DataFrameRow` produces a `view`:
```
julia> using DataFrames

julia> df = DataFrame(x=1:2)
2×1 DataFrame
 Row │ x     
     │ Int64 
─────┼───────
   1 │     1
   2 │     2

julia> DataFrames._names(df)
1-element Vector{Symbol}:
 :x

julia> DataFrames._names(df[1, :])
1-element view(::Vector{Symbol}, Base.OneTo(1)) with eltype Symbol:
 :x
```